x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/pattern/index_check_flyout/index.tsx

```diff
@@ -0,0 +1,171 @@
+  EuiFlyoutHeader,
+  EuiSpacer,
+  EuiTab,
+  EuiTabs,
+  EuiText,
+  EuiTitle,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import React, { useCallback, useEffect, useState } from 'react';
+import moment from 'moment';
+import { useIndicesCheckContext } from '../../../contexts/indices_check_context';
+
+import { EMPTY_STAT, getDocsCount, getSizeInBytes } from '../../../helpers';
+import { MeteringStatsIndex, PatternRollup } from '../../../types';
+import { useDataQualityContext } from '../../data_quality_context';
+import { IndexProperties } from '../../index_properties';
+import { CHECK_NOW } from '../../summary_table/translations';
+import { getIlmPhase } from '../helpers';
+import { IndexResultBadge } from '../../index_result_badge';
+import { HISTORY, LATEST_CHECK } from '../translations';
+import { useCurrentWindowWidth } from '../../../use_current_window_width';
+
+export interface Props {
+  ilmExplain: Record<string, IlmExplainLifecycleLifecycleExplain> | null;
+  indexName: string;
+  pattern: string;
+  patternRollup: PatternRollup | undefined;
+  stats: Record<string, MeteringStatsIndex> | null;
+  onClose: () => void;
+}
+
+export const IndexCheckFlyoutComponent: React.FC<Props> = ({
+  ilmExplain,
+  indexName,
+  pattern,
+  patternRollup,
+  stats,
+  onClose,
+}) => {
+  const currentWindowWidth = useCurrentWindowWidth();
+  const isLargeScreen = currentWindowWidth > 1720;
+  const isMediumScreen = currentWindowWidth > 1200;
+  const { httpFetch, formatBytes, formatNumber, isILMAvailable } = useDataQualityContext();
+  const { checkState, checkIndex } = useIndicesCheckContext();
+  const indexCheckState = checkState[indexName];
+  const isChecking = indexCheckState?.isChecking ?? false;
+  const partitionedFieldMetadata = indexCheckState?.partitionedFieldMetadata ?? null;
+  const indexResult = patternRollup?.results?.[indexName];
+  const [selectedTabId, setSelectedTabId] = useState(LATEST_CHECK);
+  const indexCheckFlyoutTitleId = useGeneratedHtmlId({
+    prefix: 'indexCheckFlyoutTitle',
+  });
+  const abortControllerRef = React.useRef(new AbortController());
+
+  const handleCheckNow = useCallback(() => {
+    checkIndex({
+      abortController: abortControllerRef.current,
+      indexName,
+      pattern,
+      httpFetch,
+      formatBytes,
+      formatNumber,
+    });
+  }, [checkIndex, formatBytes, formatNumber, httpFetch, indexName, pattern]);
+
+  useEffect(() => {
+    const abortController = abortControllerRef.current;
+    return () => {
+      abortController.abort();
+    };
+  }, []);
+
+  const tabs = [
+    {
+      id: LATEST_CHECK,
+      name: LATEST_CHECK,
+    },
+    {
+      id: HISTORY,
+      name: HISTORY,
+      disabled: true,
+    },
+  ];
+
+  const renderTabs = tabs.map((tab, index) => (
+    <EuiTab
+      onClick={() => setSelectedTabId(tab.id)}
+      isSelected={tab.id === selectedTabId}
+      key={index}
+      disabled={tab.disabled}
+    >
+      {tab.name}
+    </EuiTab>
+  ));
```

Comment:
What do you think about using `useMemo` for `tabs` and `useCallback` for `renderTabs` ? 
So in this way helps with performance by ensuring that tabs and renderTabs are only recalculated when necessary, reducing unnecessary re-renders

Reply:
Hey @agusruidiazgd thanks for your suggestion.

What do you think about:

1. Extracting `tabs` out of component, because it has no dependencies and has no reason to be in the render cycle.
2. Use `useMemo()` for `renderTabs` to cache it. `useCallback` here will not help, because identity of mapCallback has no affect on return value, you will still do the `map`ping part over and over again, if we talk about performance, so we need to memoize the return value not the function itself.

Let me know wdyt about this.

Reply:
perfect!